thorlcr/thorutil/thbuf.cpp

```diff
@@ -2686,11 +2713,8 @@ class CSharedFullSpillingWriteAhead : public CInterfaceOf<ISharedRowStreamReader
     }
     virtual void reset() override
     {
-        if (outputStream) // should have already been closed when inputs all stopped
-        {
-            closeWriter();
-            iFile->remove();
-        }
+        closeWriter();
+        tempFileOwner->noteSize(0);
```

Comment:
Has the file actually be removed/length set to 0?

Reply:
Agree. See my other comment reply. I think you either need to clear and recreate this object, or add a clear/set.
In standard case, CFileOwner should track the file until createOutputStream is called, the file doesn't exist until then, so ~CFileOwner is actually going to [silently] fail in it's call to iFile->remove() at the moment if this splitter hasn't spilt.